api/v1/intrusiondetection_types.go

```diff
@@ -27,8 +27,42 @@ type IntrusionDetectionSpec struct {
 	// Only DeepPacketInspection is supported for this spec.
 	// +optional
 	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
+
+	// AnomalyDetectionSpec provides configuration for running AnomalyDetectionSpec Component within
+	// IntrusionDetection
+	// +optional
+	AnomalyDetectionSpec AnomalyDetectionSpec `json:"anomalyDetection,omitempty"`
+}
+
+type AnomalyDetectionSpec struct {
+	// StorageType Sets the type of storage to use for storing Anomaly Detection Models. By default it will use the ephemeral
+	// emptyDir on the node AD will be deployed to.
+	// default: Ephemeral
+	// +optional
+	// +kubebuilder:validation:Enum=Ephemeral;Persistent
+	StorageType StorageType `json:"storageType,omitempty"`
+
+	// StorageClassName will populate the PersistentVolumeClaim.StorageClassName that is used to provision disks for the
+	// Anomaly Detection API pod for model storage. The StorageClassName should only be modified when no StorageClass is currently
+	// active. We recommend choosing a storage class dedicated to AnomalyDetection only. Otherwise, model retention
+	// cannot be guaranteed during upgrades. See https://docs.tigera.io/maintenance/upgrading for up-to-date instructions.
+	// Default: tigera-anomaly-detection-storage
+	// +optional
+	StorageClassName string `json:"storageClassName,omitempty"`
 }
 
+// StorageType is sets the type of storage to be used for the specified component.
+type StorageType string
+
+const (
+	// ephemeral storage type sets the ephemeral emptyDir() to be used by the component. Data created in this storage type will
+	// follow the Pod's lifetime and get created and deleted along with the Pod.
+	EphemeralStorageType StorageType = "Ephemeral"
+	// Persistent storage type sets the will attempt to query for the StorageClass specified by StorageClassName created for
```

Comment:
PersistentStorageType mounts a PersistentVolume of the provided StorageClassName to Anomaly Detection pods in order to store data.